stable_baselines/common/distributions.py

```diff
@@ -280,20 +280,43 @@ def sample_dtype(self):
 
 
 class CategoricalProbabilityDistribution(ProbabilityDistribution):
-    def __init__(self, logits):
+    def __init__(self, logits, action_mask=None):
         """
         Probability distributions from categorical input
 
         :param logits: ([float]) the categorical logits input
         """
         self.logits = logits
+        if action_mask is None:
+            with tf.variable_scope("input", reuse=False):
+                no_mask = tf.ones_like(self.logits)
+                self._action_mask_ph = tf.placeholder_with_default(no_mask, shape=self.logits.shape, name="action_ph")
+        else:
+            self._action_mask_ph = action_mask
         super(CategoricalProbabilityDistribution, self).__init__()
 
+    @property
+    def action_mask_ph(self):
+        return self._action_mask_ph
+
+    @property
+    def confusing_action_mask_ph(self):
+        """
+        Remap the values of the action mask, replacing 0s with -np.inf, and 1s with 0s.
+        """
+        negative_inf_vector = tf.ones_like(self._action_mask_ph, dtype=tf.float32) * -np.inf
+        zero_vector = tf.zeros_like(self._action_mask_ph, dtype=tf.float32)
+        _action_mask_ph = tf.where(tf.cast(self._action_mask_ph, dtype=tf.bool), zero_vector, negative_inf_vector)
+        return _action_mask_ph
+
```

Comment:
I have a naming obstacle about the confusing action mask vector. Is there any suggested name?